pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1060,7 +1060,37 @@ func (c *Cluster) startDeployments(mons []*monConfig, requireAllInQuorum bool) e
 			requireAllInQuorum = true
 		}
 	}
-	return c.waitForMonsToJoin(mons, requireAllInQuorum)
+	err = c.waitForMonsToJoin(mons, requireAllInQuorum)
+
+	// Check for the rare case of an extra mon deployment that needs to be cleaned up
+	c.checkForExtraMonResources(mons, deployments.Items)
+	return err
+}
+
+func (c *Cluster) checkForExtraMonResources(mons []*monConfig, deployments []apps.Deployment) string {
+	if len(deployments) <= c.spec.Mon.Count || len(deployments) <= len(mons) {
+		return ""
+	}
+
+	logger.Infof("there is an extra mon deployment that is not needed and not in quorum")
+	for _, deploy := range deployments {
+		monName := deploy.Labels[controller.DaemonIDLabel]
```

Comment:
controller.DaemonIDLabel
Here how we decide the extra one using label?

Reply:
I'll clarify the comments. Basically, if we find an extra mon deployment that is not in the ceph mon quorum, we can delete the extra mon deployment.

Reply:
How this is identified as extra mon
`monName := deploy.Labels[controller.DaemonIDLabel]`

Reply:
That's the name of the mon daemon, found in a running deployment. Then if the mon daemon of the same name is not found in the loop below comparing against each mon in the desired list of mons, it will be considered extra and needs to be removed on line 1090.